src/Main.ts

```diff
@@ -526,6 +533,23 @@ export class Main {
 
         const room = this.rooms.getByMatrixRoomId(ev.room_id);
         if (!room) {
+            const adminRoomUser = await this.datastore.getUserForAdminRoom(ev.room_id);
```

Comment:
took me a while to understand this whole section, a useful refactor could be to subdivide `onMatrixEvent` into several smaller methods and call them appropriately, rather than falling through.